pkg/cache/cluster.go

```diff
@@ -319,11 +336,13 @@ func (c *clusterCache) Invalidate(opts ...UpdateSettingsFunc) {
 }
 
 func (c *clusterCache) synced() bool {
```

Comment:
Would it make sense to make this a method on `clusterCacheSync` and move locking here?

Reply:
Done... I wanted to avoid accidental [copying of the mutex](https://eli.thegreenplace.net/2018/beware-of-copying-mutexes-in-go/), but it's true this way is cleaner.

Reply:
Done, definitely an improvement.

Reply:
Done... I have moved `synced()` into a method on `clusterCacheSync`, but there are still cases where the clusterCacheSync lock needs be held outside this method.